### ISSUE TITLE
Hasarlı Ürün Tespiti

### ISSUE THREAD

**Issue by sinemaydem** [sinemaydem/Goruntu_Isleme#1]
import cv2
import numpy as np
import os
from sklearn.svm import SVC

def load_images_from_folder(folder):
    images = []
    for filename in os.listdir(folder):
        img = cv2.imread(os.path.join(folder, filename))
        if img is not None:
            images.append(img)
    return images

def convert_to_gray(images):
    gray_images = []
    for img in images:
        gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
        gray_images.append(gray)
    return gray_images

def detect_edges(images):
    edges_images = []
    for img in images:
        edges = cv2.Canny(img, 100, 200)  
        edges_images.append(edges)
    return edges_images

def extract_features(images):
    features = []
    for img in images:
        hist = cv2.calcHist([img], [0, 1, 2], None, [256, 256, 256], [0, 256, 0, 256, 0, 256])
        hist = cv2.normalize(hist, hist)
        feature = hist.flatten()
        features.append(feature)
    return features

def train_model(features, labels):
    model = SVC(kernel='linear')
    model.fit(features, labels)
    return model

def evaluate_model(model, test_features, test_labels):
    accuracy = model.score(test_features, test_labels)
    return accuracy

normal_images = cv2.imread('anahtar_orj.png')
faulty_images = cv2.imread('anahtar_copy.png')

normal_gray_images = convert_to_gray(normal_images)
faulty_gray_images = convert_to_gray(faulty_images)

normal_edges = detect_edges(normal_gray_images)
faulty_edges = detect_edges(faulty_gray_images)

normal_features = extract_features(normal_edges)
faulty_features = extract_features(faulty_edges)

normal_labels = np.zeros(len(normal_features))
faulty_labels = np.ones(len(faulty_features))

train_features = np.concatenate((normal_features, faulty_features), axis=0)
train_labels = np.concatenate((normal_labels, faulty_labels), axis=0)

model = train_model(train_features, train_labels)

test_normal_images = cv2.imread('anahtar_orj.png')
test_faulty_images = cv2.imread('anahtar_soft.png')

test_normal_gray_images = convert_to_gray(test_normal_images)
test_faulty_gray_images = convert_to_gray(test_faulty_images)

test_normal_edges = detect_edges(test_normal_gray_images)
test_faulty_edges = detect_edges(test_faulty_gray_images)

test_normal_features = extract_features(test_normal_edges)
test_faulty_features = extract_features(test_faulty_edges)

test_normal_labels = np.zeros(len(test_normal_features))
test_faulty_labels = np.ones(len(test_faulty_features))

test_features = np.concatenate((test_normal_features, test_faulty_features), axis=0)
test_labels = np.concatenate((test_normal_labels, test_faulty_labels), axis=0)

accuracy = evaluate_model(model, test_features, test_labels)
print("Model accuracy:", accuracy)
